### ISSUE TITLE
`TypedDict` fields should be considered invariant

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6190]
Pyright doesn't enforce that fields within a `TypedDict` are invariant, but it should because fields are mutable.

```python
from typing import TypedDict

class TD1(TypedDict):
    a: int

class TD2(TypedDict):
    a: float

td1: TD1 = TD1(a=1)

# This should generate an error but does not.
td2: TD2 = td1
```

**Comment by tejasshaha07** [comment]
Can we use a data class along with a type checker? This will generate error? 
TypedDict has a subset of the keys and types defined in another TypedDict. This is why assigning td1 to td2 doesn't generate an error.

**Comment by erictraut** [comment]
This is included in pyright 1.1.333, which I just published. It will also be included in a future release of pylance.
